lib/modules.nix

```
@@ -1424,20 +1424,118 @@ let
   # as a definition, as the new definition will not keep the mkOverride /
   # mkDefault properties of the previous option.
   #
-  mkAliasDefinitions = mkAliasAndWrapDefinitions id;
-  mkAliasAndWrapDefinitions = wrap: option: mkAliasIfDef option (wrap (mkMerge option.definitions));
+  mkAliasDefinitions = mkAliasDefinitionsWith {
+    withPriority = false;
+    # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+    withLocation = false;
+  };
+
+  mkAliasAndWrapDefinitions =
+    wrapper:
+    mkAliasDefinitionsWith {
+      inherit wrapper;
+      withPriority = false;
+      # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+      withLocation = false;
+    };
 
   # Similar to mkAliasAndWrapDefinitions but copies over the priority from the
   # option as well.
   #
   # If a priority is not set, it assumes a priority of defaultOverridePriority.
   mkAliasAndWrapDefsWithPriority =
-    wrap: option:
+    wrapper:
+    mkAliasDefinitionsWith {
+      inherit wrapper;
+      # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+      withLocation = false;
+    };
+
+  /**
+    A function which transfers all definitions from one option to another.
+    This is useful for renaming options, and also for including properties from
+    another module system, including sub-modules.
+
+    Naturally, definitions are only transferred if the option is defined.
+    Supplying a non-option value is also silently ignored, which is useful for
+    referencing options which may not exist.
+
+    :::{.note}
+
+    This is different than taking the value of the option and using it
+    as a definition, as the new definition will not keep the `mkOverride` /
+    `mkDefault` properties of the previous option.
+
+    If the option you are copying does not have nested properties, or you need
+    to perform some transformation to the value, you may prefer `mkDerivedConfig`,
+    which preserves the option's _overall_ override priority.
+
+    :::
+
+    :::{.caution}
+
+    Transferring definitions to an option with a different `type` may result in
+    the unexpected merge semantics.
+
+    :::
+
+    # Inputs
+
+    `config` (AttrSet)
+
+    : `wrapper` (AttrSet -> Any)
+
+      : A function to wrap the final `mkMerge` definition
+        (e.g. `setAttrByPath [ "foo" "bar" ]`) (default `lib.id`)
+
+      `withPriority` (Bool)
+
+      : Whether to wrap definition with the option's overall priority (`true` by default)
+
+      `withLocation` (Bool)
+
+      : Whether to preserve definitions original file location (`true` by default)
+
+    `option` (Option or Any)
+
+    : The option from which to transfer definitions, if defined.
```

```suggestion
    : The option from which to transfer definitions, if defined.
      Non-option values are silently ignored.
```

I doubt that this is a good thing.
Could we make this an error for direct invocations of this new function?
I *think* a better behavior would be to require an "option tree", which is an actual option, or an attrset of option trees. (Similar to how we interpret `options` in the module syntax, but for evaluated options in this case)